### PR TITLE
add support for deleting property

### PR DESCRIPTION
**Pull request by acquamarin**
1. This PR adds support for deleting node/rel propties. (Without supporting transaction).
2. Refactors the copy_csv/drop_table operators.
3. Breaks the assumption that the propertyID must be consecutive of a table since we allow user removing a property from a table now.

### CODE REVIEW

File: src/include/binder/bind/bound_drop_column.h [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include "binder/bound_statement.h"
+
+namespace kuzu {
+namespace binder {
+
+class BoundDropColumn : public BoundStatement {
```
Comment (andyfengHKU) [inline src/include/binder/bind/bound_drop_column.h:8]: Consider adding a BoundDDL base class for all DDL statements.

A separate comments that is not related to this PR. I notice the folder name seems to be wrong. It should "ddl" rather than "bind". Can u mimic the structure as in the parser module to rename this folder to "ddl" and "copy_csv"?

File: src/include/common/statement_type.h [hunk 2]
```diff
@@ -11,14 +11,16 @@ enum class StatementType : uint8_t {
     CREATE_REL_CLAUSE = 2,
     COPY_CSV = 3,
     DROP_TABLE = 4,
+    DROP_COLUMN = 5,
```
Comment (andyfengHKU) [inline src/include/common/statement_type.h:14]: DROP_PROPERTY

File: src/binder/expression_binder.cpp [hunk 3]
```diff
@@ -351,8 +351,7 @@ shared_ptr<Expression> ExpressionBinder::bindNodeLabelFunction(const Expression&
     nodeLabels.resize(maxNodeTableID + 1);
     for (auto i = 0; i < nodeLabels.size(); ++i) {
         if (catalogContent->containNodeTable(i)) {
-            auto tableSchema = catalogContent->getNodeTableSchema(i);
-            nodeLabels[i] = Literal(tableSchema->tableName);
+            nodeLabels[i] = Literal(catalogContent->getTableSchema(i)->tableName);
```
Comment (andyfengHKU) [inline src/binder/expression_binder.cpp:354]: for consistency either always use getTableName() or use getTableSchema->tableName

File: src/include/catalog/catalog_structs.h [hunk 4]
```diff
@@ -58,30 +58,47 @@ struct Property : PropertyNameDataType {
...
+            [&propertyName](const Property& property) { return property.name == propertyName; });
+    }
+
+    string getPropertyName(property_id_t propertyID) const;
+
 public:
     string tableName;
     table_id_t tableID;
     bool isNodeTable;
+    vector<Property> properties;
```
Comment (andyfengHKU) [inline src/include/catalog/catalog_structs.h:92]: add an issue about replacing isNodeTable boolean field with a table type enum. Enum is always preferred over  boolean check whenever possible.

File: src/include/catalog/catalog.h [hunk 5]
```diff
@@ -51,22 +51,14 @@ class CatalogContent {
...
     }
 
-    inline NodeTableSchema* getNodeTableSchema(table_id_t tableID) const {
-        assert(containNodeTable(tableID));
-        return nodeTableSchemas.at(tableID).get();
-    }
-    inline RelTableSchema* getRelTableSchema(table_id_t tableID) const {
-        assert(containRelTable(tableID));
-        return relTableSchemas.at(tableID).get();
+    TableSchema* getTableSchema(table_id_t tableID) const {
```
Comment (andyfengHKU) [inline src/include/catalog/catalog.h:58]: protect this function by adding contains check.

Also add back getNodeTableSchema() and getRelTableSchema()

File: src/binder/bind/bind_updating_clause.cpp [hunk 6]
```diff
@@ -57,7 +57,8 @@ unique_ptr<BoundCreateNode> Binder::bindCreateNode(
             "Create node " + node->getRawName() + " with multiple node labels is not supported.");
     }
     auto nodeTableID = node->getSingleTableID();
-    auto nodeTableSchema = catalog.getReadOnlyVersion()->getNodeTableSchema(nodeTableID);
+    auto nodeTableSchema =
```
Comment (andyfengHKU) [inline src/binder/bind/bind_updating_clause.cpp:60]: revert

File: src/binder/bind/bind_updating_clause.cpp [hunk 7]
```diff
@@ -180,7 +181,8 @@ unique_ptr<BoundDeleteNode> Binder::bindDeleteNode(shared_ptr<NodeExpression> no
             "Delete node " + node->getRawName() + " with multiple node labels is not supported.");
     }
     auto nodeTableID = node->getSingleTableID();
-    auto nodeTableSchema = catalog.getReadOnlyVersion()->getNodeTableSchema(nodeTableID);
+    auto nodeTableSchema =
```
Comment (andyfengHKU) [inline src/binder/bind/bind_updating_clause.cpp:184]: revert

File: src/binder/bind/bind_graph_pattern.cpp [hunk 8]
```diff
@@ -209,7 +211,8 @@ shared_ptr<NodeExpression> Binder::createQueryNode(const NodePattern& nodePatter
     // resolve properties associate with node table
     vector<NodeTableSchema*> nodeTableSchemas;
     for (auto tableID : tableIDs) {
-        nodeTableSchemas.push_back(catalog.getReadOnlyVersion()->getNodeTableSchema(tableID));
+        nodeTableSchemas.push_back(
```
Comment (andyfengHKU) [inline src/binder/bind/bind_graph_pattern.cpp:214]: revert

File: src/binder/bind/bind_graph_pattern.cpp [hunk 9]
```diff
@@ -131,7 +132,8 @@ void Binder::bindQueryRel(const RelPattern& relPattern, const shared_ptr<NodeExp
     // resolve properties associate with rel table
     vector<RelTableSchema*> relTableSchemas;
     for (auto tableID : tableIDs) {
-        relTableSchemas.push_back(catalog.getReadOnlyVersion()->getRelTableSchema(tableID));
+        relTableSchemas.push_back(
```
Comment (andyfengHKU) [inline src/binder/bind/bind_graph_pattern.cpp:135]: revert

File: src/binder/bind/bind_ddl.cpp [hunk 10]
```diff
@@ -52,9 +54,20 @@ unique_ptr<BoundStatement> Binder::bindDropTable(const Statement& statement) {
     if (isNodeTable) {
         validateNodeTableHasNoEdge(catalog, tableID);
     }
-    return make_unique<BoundDropTable>(
-        isNodeTable ? (TableSchema*)catalogContent->getNodeTableSchema(tableID) :
-                      (TableSchema*)catalogContent->getRelTableSchema(tableID));
+    return make_unique<BoundDropTable>(tableID, tableName);
+}
+
+unique_ptr<BoundStatement> Binder::bindDropColumn(const Statement& statement) {
```
Comment (andyfengHKU) [inline src/binder/bind/bind_ddl.cpp:60]: bindDropProperty

File: src/binder/bind/bind_ddl.cpp [hunk 11]
```diff
@@ -52,9 +54,20 @@ unique_ptr<BoundStatement> Binder::bindDropTable(const Statement& statement) {
...
+    auto& dropColumn = (DropColumn&)statement;
+    auto tableName = dropColumn.getTableName();
+    validateTableExist(catalog, tableName);
+    auto catalogContent = catalog.getReadOnlyVersion();
+    auto isNodeTable = catalogContent->containNodeTable(tableName);
+    auto tableID = isNodeTable ? catalogContent->getNodeTableIDFromName(tableName) :
```
Comment (andyfengHKU) [inline src/binder/bind/bind_ddl.cpp:66]: Up to you but you can have an addition API in catalog getTableID(tableName) to which you push the check of isNodeTable.

File: src/catalog/catalog_structs.cpp [hunk 12]
```diff
@@ -37,6 +38,16 @@ string getRelMultiplicityAsString(RelMultiplicity relMultiplicity) {
     }
 }
 
+string TableSchema::getPropertyName(property_id_t propertyID) const {
+    for (auto& property : properties) {
```
Comment (andyfengHKU) [inline src/catalog/catalog_structs.cpp:42]: consider indexing properties in TableSchema

File: src/main/connection.cpp [hunk 13]
```diff
@@ -132,7 +133,9 @@ string Connection::getNodePropertyNames(const string& tableName) {
     string result = tableName + " properties: \n";
     auto tableID = catalog->getReadOnlyVersion()->getNodeTableIDFromName(tableName);
     auto primaryKeyPropertyID =
-        catalog->getReadOnlyVersion()->getNodeTableSchema(tableID)->getPrimaryKey().propertyID;
+        ((NodeTableSchema*)catalog->getReadOnlyVersion()->getTableSchema(tableID))
```
Comment (andyfengHKU) [inline src/main/connection.cpp:136]: revert

File: src/planner/join_order_enumerator.cpp [hunk 14]
```diff
@@ -172,7 +172,7 @@ static shared_ptr<Expression> extractIndexExpression(Expression& expression) {
 static pair<shared_ptr<Expression>, expression_vector> splitIndexAndPredicates(
     const CatalogContent& catalogContent, NodeExpression& node,
     const expression_vector& predicates) {
-    auto nodeTableSchema = catalogContent.getNodeTableSchema(node.getSingleTableID());
+    auto nodeTableSchema = (NodeTableSchema*)catalogContent.getTableSchema(node.getSingleTableID());
```
Comment (andyfengHKU) [inline src/planner/join_order_enumerator.cpp:175]: revert

File: src/include/planner/logical_plan/logical_operator/logical_drop_column.h [hunk 15]
```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include "base_logical_operator.h"
+
+namespace kuzu {
+namespace planner {
+
+class LogicalDropColumn : public LogicalOperator {
```
Comment (andyfengHKU) [inline src/include/planner/logical_plan/logical_operator/logical_drop_column.h:8]: LogicalDropProperty

File: src/include/processor/mapper/plan_mapper.h [hunk 16]
```diff
@@ -70,6 +70,7 @@ class PlanMapper {
         LogicalOperator* logicalOperator);
     unique_ptr<PhysicalOperator> mapLogicalCopyCSVToPhysical(LogicalOperator* logicalOperator);
     unique_ptr<PhysicalOperator> mapLogicalDropTableToPhysical(LogicalOperator* logicalOperator);
+    unique_ptr<PhysicalOperator> mapLogicalDropColumnToPhysical(LogicalOperator* logicalOperator);
```
Comment (andyfengHKU) [inline src/include/processor/mapper/plan_mapper.h:73]: mapLogicalDropPropertyToPhysical

File: src/include/processor/operator/copy_csv/copy_csv.h [hunk 17]
```diff
@@ -30,19 +28,20 @@ class CopyCSV : public PhysicalOperator {
...
         auto numTuples = tablesStatistics->getReadOnlyVersion()
-                             ->tableStatisticPerTable.at(tableSchema.tableID)
+                             ->tableStatisticPerTable.at(tableID)
                              ->getNumTuples();
         if (numTuples > 0) {
+            auto tableName = catalog->getReadOnlyVersion()->getTableSchema(tableID)->tableName;
             throw CopyCSVException(
                 "COPY CSV commands can be executed only on completely empty tables. Table: " +
-                tableSchema.tableName + " has " + to_string(numTuples) + " many tuples.");
+                tableName + " has " + to_string(numTuples) + " many tuples.");
```
Comment (andyfengHKU) [inline src/include/processor/operator/copy_csv/copy_csv.h:37]: Up to u but I guess we don't need to print tableName in return/error msg. After all, the msg points to a specific ddl/copyCSV statement which doesn't have any ambiguity. This comment applies to all DDL and CopyCSV operators.

If so, you can get rid of the code `auto tableName = catalog->getReadOnlyVersion()->getTableSchema(tableID)->tableName;`

File: src/include/processor/operator/physical_operator.h [hunk 18]
```diff
@@ -20,6 +20,7 @@ enum class PhysicalOperatorType : uint8_t {
     CROSS_PRODUCT,
     DELETE_NODE,
     DELETE_REL,
+    DROP_COLUMN,
```
Comment (andyfengHKU) [inline src/include/processor/operator/physical_operator.h:23]: DROP_PROPERTY

File: src/processor/operator/ddl/CMakeLists.txt [hunk 19]
```diff
@@ -1,7 +1,8 @@
 add_library(kuzu_processor_operator_ddl
         OBJECT
         create_node_table.cpp
-        create_rel_table.cpp)
+        create_rel_table.cpp
+        drop_column.cpp)
```
Comment (andyfengHKU) [inline src/processor/operator/ddl/CMakeLists.txt:5]: rename file as well

File: src/processor/operator/ddl/create_node_table.cpp [hunk 20]
```diff
@@ -6,7 +6,7 @@ namespace processor {
 string CreateNodeTable::execute() {
     auto newTableID = catalog->addNodeTableSchema(tableName, primaryKeyIdx, propertyNameDataTypes);
     nodesStatisticsAndDeletedIDs->addNodeStatisticsAndDeletedIDs(
-        catalog->getWriteVersion()->getNodeTableSchema(newTableID));
+        (NodeTableSchema*)catalog->getWriteVersion()->getTableSchema(newTableID));
```
Comment (andyfengHKU) [inline src/processor/operator/ddl/create_node_table.cpp:9]: revert

File: src/processor/operator/ddl/create_rel_table.cpp [hunk 21]
```diff
@@ -6,7 +6,7 @@ namespace processor {
 string CreateRelTable::execute() {
     auto newRelTableID = catalog->addRelTableSchema(
         tableName, relMultiplicity, propertyNameDataTypes, srcDstTableIDs);
-    relsStatistics->addTableStatistic(catalog->getWriteVersion()->getRelTableSchema(newRelTableID));
+    relsStatistics->addTableStatistic(catalog->getWriteVersion()->getTableSchema(newRelTableID));
```
Comment (andyfengHKU) [inline src/processor/operator/ddl/create_rel_table.cpp:9]: One thing worth noticing here is you don't need to cast when adding relStatistics but do need cast for nodeStatistics. This might be desired behaviour but worth double checking

File: src/storage/in_mem_csv_copier/in_mem_arrow_node_copier.cpp [hunk 22]
```diff
@@ -11,7 +11,7 @@ InMemArrowNodeCopier::InMemArrowNodeCopier(CSVDescription& csvDescription, strin
     NodesStatisticsAndDeletedIDs* nodesStatisticsAndDeletedIDs)
     : InMemStructuresCopier{csvDescription, std::move(outputDirectory), taskScheduler, catalog},
       numNodes{UINT64_MAX}, nodesStatisticsAndDeletedIDs{nodesStatisticsAndDeletedIDs} {
-    nodeTableSchema = catalog.getReadOnlyVersion()->getNodeTableSchema(tableID);
+    nodeTableSchema = (NodeTableSchema*)catalog.getReadOnlyVersion()->getTableSchema(tableID);
```
Comment (andyfengHKU) [inline src/storage/in_mem_csv_copier/in_mem_arrow_node_copier.cpp:14]: revert

File: src/include/storage/store/nodes_store.h [hunk 23]
```diff
@@ -33,8 +33,9 @@ class NodesStore {
     // nodeStore when checkpointing and not in recovery mode.
     inline void createNodeTable(
         table_id_t tableID, BufferManager* bufferManager, WAL* wal, Catalog* catalog) {
-        nodeTables[tableID] = make_unique<NodeTable>(&nodesStatisticsAndDeletedIDs, *bufferManager,
-            isInMemoryMode, wal, catalog->getReadOnlyVersion()->getNodeTableSchema(tableID));
+        nodeTables[tableID] =
+            make_unique<NodeTable>(&nodesStatisticsAndDeletedIDs, *bufferManager, isInMemoryMode,
+                wal, (NodeTableSchema*)catalog->getReadOnlyVersion()->getTableSchema(tableID));
```
Comment (andyfengHKU) [inline src/include/storage/store/nodes_store.h:38]: revert

File: src/include/storage/store/node_table.h [hunk 24]
```diff
@@ -29,31 +29,28 @@ class NodeTable {
     void scan(Transaction* transaction, const shared_ptr<ValueVector>& inputIDVector,
         const vector<uint32_t>& columnIdxes, vector<shared_ptr<ValueVector>> outputVectors);
 
-    inline Column* getPropertyColumn(uint64_t propertyIdx) {
-        return propertyColumns[propertyIdx].get();
+    inline Column* getPropertyColumn(property_id_t propertyIdx) {
+        return propertyColumns.at(propertyIdx).get();
```
Comment (andyfengHKU) [inline src/include/storage/store/node_table.h:33]: assert contains

File: src/storage/in_mem_csv_copier/in_mem_rel_csv_copier.cpp [hunk 25]
```diff
@@ -14,12 +14,14 @@ InMemRelCSVCopier::InMemRelCSVCopier(CSVDescription& csvDescription, string outp
       maxNodeOffsetsPerTable{move(maxNodeOffsetsPerNodeTable)}, relsStatistics{relsStatistics} {
     dummyReadOnlyTrx = Transaction::getDummyReadOnlyTrx();
     startRelID = relsStatistics->getNextRelID(dummyReadOnlyTrx.get());
-    relTableSchema = catalog.getReadOnlyVersion()->getRelTableSchema(tableID);
+    relTableSchema = (RelTableSchema*)catalog.getReadOnlyVersion()->getTableSchema(tableID);
```
Comment (andyfengHKU) [inline src/storage/in_mem_csv_copier/in_mem_rel_csv_copier.cpp:17]: revert and check through this file

File: src/storage/store/node_table.cpp [hunk 26]
```diff
@@ -49,7 +47,7 @@ node_offset_t NodeTable::addNodeAndResetProperties(ValueVector* primaryKeyVector
                            primaryKeyVector->getValue<ku_string_t>(pkValPos).getAsString();
         throw RuntimeException(Exception::getExistedPKExceptionMsg(pkStr));
     }
-    for (auto& column : propertyColumns) {
+    for (auto& [propertyID, column] : propertyColumns) {
```
Comment (andyfengHKU) [inline src/storage/store/node_table.cpp:50]: `auto& [_, column]` if you don't need propertyID

File: src/storage/store/rel_table.cpp [hunk 27]
```diff
@@ -171,7 +169,7 @@ void DirectedRelTableData::deleteRel(
     auto nodeOffset =
         boundVector->readNodeOffset(boundVector->state->selVector->selectedPositions[0]);
     adjColumn->setNodeOffsetToNull(nodeOffset);
-    for (auto& propertyColumn : propertyColumns.at(boundTableID)) {
+    for (auto& [propertyID, propertyColumn] : propertyColumns.at(boundTableID)) {
```
Comment (andyfengHKU) [inline src/storage/store/rel_table.cpp:172]: propertyID -> _. Here and everywhere

File: src/storage/store/rel_table.cpp [hunk 28]
```diff
@@ -192,17 +190,17 @@ vector<unique_ptr<ListsUpdateIterator>> DirectedRelTableData::getListsUpdateIter
     vector<unique_ptr<ListsUpdateIterator>> listsUpdateIterators;
     listsUpdateIterators.push_back(
         ListsUpdateIteratorFactory::getListsUpdateIterator(adjLists.at(boundNodeTableID).get()));
-    for (auto& propList : propertyLists.at(boundNodeTableID)) {
-        listsUpdateIterators.push_back(
-            ListsUpdateIteratorFactory::getListsUpdateIterator(propList.get()));
+    for (auto i = 0u; i < propertyLists.at(boundNodeTableID).size(); i++) {
```
Comment (andyfengHKU) [inline src/storage/store/rel_table.cpp:193]: For consistency, better do `auto& [, propertyList] : propertyLists.at(boundNodeTableID)`

File: src/storage/store/rel_table.cpp [hunk 29]
```diff
@@ -192,17 +190,17 @@ vector<unique_ptr<ListsUpdateIterator>> DirectedRelTableData::getListsUpdateIter
...
+            propertyLists.at(boundNodeTableID).at(i).get()));
     }
     return listsUpdateIterators;
 }
 
 RelTable::RelTable(const Catalog& catalog, table_id_t tableID, BufferManager& bufferManager,
     MemoryManager& memoryManager, bool isInMemoryMode, WAL* wal)
     : tableID{tableID}, wal{wal} {
-    auto tableSchema = catalog.getReadOnlyVersion()->getRelTableSchema(tableID);
+    auto tableSchema = (RelTableSchema*)catalog.getReadOnlyVersion()->getTableSchema(tableID);
```
Comment (andyfengHKU) [inline src/storage/store/rel_table.cpp:203]: revert

File: src/storage/wal_replayer.cpp [hunk 30]
```diff
@@ -206,7 +206,8 @@ void WALReplayer::replayWALRecord(WALRecord& walRecord) {
         if (isCheckpoint) {
             auto tableID = walRecord.copyNodeCsvRecord.tableID;
             if (!isRecovering) {
-                auto nodeTableSchema = catalog->getReadOnlyVersion()->getNodeTableSchema(tableID);
+                auto nodeTableSchema =
```
Comment (andyfengHKU) [inline src/storage/wal_replayer.cpp:209]: check this file and revert

File: src/storage/wal_replayer_utils.cpp [hunk 31]
```diff
@@ -7,7 +7,7 @@ namespace storage {
 
 void WALReplayerUtils::createEmptyDBFilesForNewRelTable(Catalog* catalog, table_id_t tableID,
     const string& directory, const map<table_id_t, uint64_t>& maxNodeOffsetsPerTable) {
-    auto relTableSchema = catalog->getReadOnlyVersion()->getRelTableSchema(tableID);
+    auto relTableSchema = (RelTableSchema*)catalog->getReadOnlyVersion()->getTableSchema(tableID);
```
Comment (andyfengHKU) [inline src/storage/wal_replayer_utils.cpp:10]: ditto

File: src/include/parser/ddl/drop_column.h [hunk 32]
```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include "parser/ddl/ddl.h"
+
+namespace kuzu {
+namespace parser {
+
+using namespace std;
+
+class DropColumn : public DDL {
```
[the remark this answers is not in the document]
Comment (acquamarin) [reply src/include/parser/ddl/drop_column.h:10]: Renamed

File: src/include/parser/ddl/drop_column.h [hunk 33]
```diff
@@ -0,0 +1,23 @@
...
+public:
+    explicit DropColumn(string tableName, string columnName)
+        : DDL{StatementType::DROP_COLUMN, std::move(tableName)}, columnName{std::move(columnName)} {
+    }
+
+    inline string getPropertyName() const { return columnName; };
+
+private:
+    string columnName;
```
[the remark this answers is not in the document]
Comment (acquamarin) [reply src/include/parser/ddl/drop_column.h:19]: renamed to propertyName

File: src/include/binder/bind/bound_drop_column.h (hunk 1 above)
Comment (acquamarin) [reply src/include/binder/bind/bound_drop_column.h:8]: Refactored the structure of the bound ddl folder. BoundDDL statements don't have any common fields. 
Added a BoundDDL base class

File: src/include/common/statement_type.h (hunk 2 above)
Comment (acquamarin) [reply src/include/common/statement_type.h:14]: renamed

File: src/binder/expression_binder.cpp (hunk 3 above)
Comment (acquamarin) [reply src/binder/expression_binder.cpp:354]: Changed to `getTableName()`

File: src/include/catalog/catalog_structs.h (hunk 4 above)
Comment (acquamarin) [reply src/include/catalog/catalog_structs.h:92]: Opened issue: https://github.com/kuzudb/kuzu/issues/1170

File: src/binder/bind/bind_ddl.cpp (hunk 10 above)
Comment (acquamarin) [reply src/binder/bind/bind_ddl.cpp:60]: renamed

File: src/include/planner/logical_plan/logical_operator/logical_drop_column.h (hunk 15 above)
Comment (acquamarin) [reply src/include/planner/logical_plan/logical_operator/logical_drop_column.h:8]: renamed

File: src/include/processor/mapper/plan_mapper.h (hunk 16 above)
Comment (acquamarin) [reply src/include/processor/mapper/plan_mapper.h:73]: renamed

File: src/include/processor/operator/physical_operator.h (hunk 18 above)
Comment (acquamarin) [reply src/include/processor/operator/physical_operator.h:23]: renamed

File: src/processor/operator/ddl/CMakeLists.txt (hunk 19 above)
Comment (acquamarin) [reply src/processor/operator/ddl/CMakeLists.txt:5]: renamed

File: src/include/storage/store/node_table.h (hunk 24 above)
Comment (acquamarin) [reply src/include/storage/store/node_table.h:33]: Added an assertion constrains here

File: src/storage/store/node_table.cpp (hunk 26 above)
Comment (acquamarin) [reply src/storage/store/node_table.cpp:50]: changed `propertyID`->`_`

File: src/storage/store/rel_table.cpp (hunk 27 above)
Comment (acquamarin) [reply src/storage/store/rel_table.cpp:172]: done

File: src/storage/store/rel_table.cpp (hunk 28 above)
Comment (acquamarin) [reply src/storage/store/rel_table.cpp:193]: This function has been changed in my another PR, i just want to make it work in this pr

File: src/include/processor/operator/copy_csv/copy_csv.h (hunk 17 above)
Comment (acquamarin) [reply src/include/processor/operator/copy_csv/copy_csv.h:37]: I think this may be helpful for debugging. It can quickly tell you which statement/table has failed.

File: src/binder/bind/bind_ddl.cpp (hunk 11 above)
Comment (acquamarin) [reply src/binder/bind/bind_ddl.cpp:66]: It is possible for a node table and a rel table to have the exactly the same name. So i can't create a generic getTableID(tableName) function

File: src/storage/wal_replayer.cpp (hunk 30 above)
Comment (acquamarin) [reply src/storage/wal_replayer.cpp:209]: done

File: src/storage/wal_replayer_utils.cpp (hunk 31 above)
Comment (acquamarin) [reply src/storage/wal_replayer_utils.cpp:10]: done

File: src/storage/store/rel_table.cpp (hunk 29 above)
Comment (acquamarin) [reply src/storage/store/rel_table.cpp:203]: done

File: src/storage/in_mem_csv_copier/in_mem_rel_csv_copier.cpp (hunk 25 above)
Comment (acquamarin) [reply src/storage/in_mem_csv_copier/in_mem_rel_csv_copier.cpp:17]: done

File: src/include/storage/store/nodes_store.h (hunk 23 above)
Comment (acquamarin) [reply src/include/storage/store/nodes_store.h:38]: done

File: src/storage/in_mem_csv_copier/in_mem_arrow_node_copier.cpp (hunk 22 above)
Comment (acquamarin) [reply src/storage/in_mem_csv_copier/in_mem_arrow_node_copier.cpp:14]: done

File: src/processor/operator/ddl/create_rel_table.cpp (hunk 21 above)
Comment (acquamarin) [reply src/processor/operator/ddl/create_rel_table.cpp:9]: To be safe, revert to `getReltableSchema()`

File: src/processor/operator/ddl/create_node_table.cpp (hunk 20 above)
Comment (acquamarin) [reply src/processor/operator/ddl/create_node_table.cpp:9]: done

File: src/planner/join_order_enumerator.cpp (hunk 14 above)
Comment (acquamarin) [reply src/planner/join_order_enumerator.cpp:175]: done

File: src/main/connection.cpp (hunk 13 above)
Comment (acquamarin) [reply src/main/connection.cpp:136]: done

File: src/binder/bind/bind_graph_pattern.cpp (hunk 9 above)
Comment (acquamarin) [reply src/binder/bind/bind_graph_pattern.cpp:135]: done

File: src/binder/bind/bind_graph_pattern.cpp (hunk 8 above)
Comment (acquamarin) [reply src/binder/bind/bind_graph_pattern.cpp:214]: done

File: src/binder/bind/bind_updating_clause.cpp (hunk 7 above)
Comment (acquamarin) [reply src/binder/bind/bind_updating_clause.cpp:184]: done

File: src/binder/bind/bind_updating_clause.cpp (hunk 6 above)
Comment (acquamarin) [reply src/binder/bind/bind_updating_clause.cpp:60]: done

File: src/catalog/catalog_structs.cpp (hunk 12 above)
Comment (acquamarin) [reply src/catalog/catalog_structs.cpp:42]: As we discussed, let's not do it now.
